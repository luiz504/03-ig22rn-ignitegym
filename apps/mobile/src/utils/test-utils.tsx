import { NavigationContainer } from '@react-navigation/native'
import { render } from '@testing-library/react-native'
import { NativeBaseProvider } from 'native-base'
import { FC, ReactNode } from 'react'
import { THEME } from '~/theme'

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  const inset = {
    frame: { x: 0, y: 0, width: 0, height: 0 },
    insets: { top: 0, left: 0, right: 0, bottom: 0 },
  }
  return (
    <NativeBaseProvider theme={THEME} initialWindowMetrics={inset}>
      <NavigationContainer>{children}</NavigationContainer>
    </NativeBaseProvider>
  )
}

const customRender: typeof render = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react-native'

export { customRender as render }
