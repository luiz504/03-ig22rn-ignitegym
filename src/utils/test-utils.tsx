import { render } from '@testing-library/react-native'
import { NativeBaseProvider } from 'native-base'
import { FC, ReactNode } from 'react'
import { THEME } from '~/theme'

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <NativeBaseProvider theme={THEME}>{children}</NativeBaseProvider>
}

const customRender: typeof render = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react-native'

export { customRender as render }
