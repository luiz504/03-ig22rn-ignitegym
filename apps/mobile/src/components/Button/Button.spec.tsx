import { renderWithNBProviders, screen } from '~/utils/test/test-utils'
import { Button } from '.'
describe('Button Component', () => {
  const btnId = 'btn'
  const labelText = 'label'
  it('should render correctly default variant "solid"', () => {
    renderWithNBProviders(<Button label={labelText} testID={btnId} />)

    const btnElement = screen.getByTestId(btnId)
    const labelElement = screen.getByText(labelText)

    expect(btnElement.props.style).toMatchSnapshot('btn-solid-style')
    expect(labelElement.props.style).toMatchSnapshot('btn-solid-label-style')
  })

  it('should render correctly default variant "outline"', () => {
    renderWithNBProviders(
      <Button variant="outline" label={labelText} testID={btnId} />,
    )

    const btnElement = screen.getByTestId(btnId)
    const labelElement = screen.getByText(labelText)

    expect(btnElement.props.style).toMatchSnapshot('btn-outline-style')
    expect(labelElement.props.style).toMatchSnapshot('btn-outline-label-style')
  })
})
