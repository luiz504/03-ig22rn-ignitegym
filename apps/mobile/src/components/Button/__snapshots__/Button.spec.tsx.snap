// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly default variant "outline": btn-outline-label-style 1`] = `
{
  "backgroundColor": undefined,
  "color": "#00B37E",
  "fontFamily": "Roboto-Bold",
  "fontSize": 16,
  "fontStyle": "normal",
  "fontWeight": "400",
  "letterSpacing": 0,
  "lineHeight": 24,
  "textDecorationLine": undefined,
}
`;

exports[`Button Component should render correctly default variant "outline": btn-outline-style 1`] = `
{
  "alignItems": "center",
  "backgroundColor": "transparent",
  "borderColor": "#00B37E",
  "borderRadius": 4,
  "borderWidth": 1.5,
  "flexDirection": "row",
  "height": 56,
  "justifyContent": "center",
  "paddingBottom": 10,
  "paddingLeft": 12,
  "paddingRight": 12,
  "paddingTop": 10,
  "width": "100%",
}
`;

exports[`Button Component should render correctly default variant "solid": btn-solid-label-style 1`] = `
{
  "backgroundColor": undefined,
  "color": "#FFFFFF",
  "fontFamily": "Roboto-Bold",
  "fontSize": 16,
  "fontStyle": "normal",
  "fontWeight": "400",
  "letterSpacing": 0,
  "lineHeight": 24,
  "textDecorationLine": undefined,
}
`;

exports[`Button Component should render correctly default variant "solid": btn-solid-style 1`] = `
{
  "alignItems": "center",
  "backgroundColor": "#00875F",
  "borderColor": "#00875F",
  "borderRadius": 4,
  "borderWidth": 1.5,
  "flexDirection": "row",
  "height": 56,
  "justifyContent": "center",
  "paddingBottom": 10,
  "paddingLeft": 12,
  "paddingRight": 12,
  "paddingTop": 10,
  "width": "100%",
}
`;
